crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -152,6 +152,13 @@ def transition(inst, name, state)
       if node.crowbar["state"] != state
         @logger.debug("Crowbar transition: state has changed so we need to do stuff for #{name} to #{state}")
 
+        # Do not allow change to shutdown state from crowbar_upgrade
+        # (we need to reboot the nodes for upgrade, but without changing the state)
+        if node.crowbar["state"] == "crowbar_upgrade" && state == "shutdown"
+          @logger.debug("Crowbar transition: current state is crowbar_upgrade; change to #{state} not allowed")
```

Comment:
Metrics/LineLength: Line is too long. [111/100] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)